### DISCUSSION TITLE
Request to Update Commit Email Attribution Due to Inaccessible Corporate Account

### DISCUSSION THREAD

**Discussion by ArpitLT** [apache/groovy-geb#202] [General]

**Comment by ghost** [comment]
@erdi @ldaley can you please help?

**Comment by jonnybot0** [comment]
Hey, @ArpitLT. Let me discuss with the other contributors and get back to you. I'd generally like to help people keep their contribution history, but I also want to be cautious about rewriting history. :)

**Reply by jonnybot0** [reply]
So, my first question, @ArpitLT / @arpitgupta-it  is whether you've contacted GitHub account support. https://support.github.com/contact/account

Rewriting history is a bit of a nuisance for fellow committers and can hurt things like traceability. It's not the sort of thing I'd ordinarily want to do.

I think the more sensible path would be to contact Github support, since you can still log in to your old account, and see if you can get your old email added as an alternate email address. It may help if you have account recovery codes and such setup on your old account.

See, for example, this discussion: https://github.com/orgs/community/discussions/67018

**Reply by ghost** [reply]
I checked with them. However they suggested to allow geb repo maintainers do it.

If this could be done, it would be really helpful.

**Reply by jonnybot0** [reply]
Thank you @erdi. I was just about to reply that the more I think about it, the more that I don't think rewriting the history is prudent. It would mean altering release tags, which decreases traceability on the artifacts produced. I don't think that's a good move for making the project's history traceable or easy to audit.

@arpitgupta-it if you would like to point Github's support to this conversation as proof that the Geb maintainers won't rewrite history for you, and so you need their help to update the email addresses on your account, feel free.

If possible, you could also contact your former workplace to see if they would be willing to temporarily reactivate your email address. I realize that may be a longshot, but it would be a way to migrate your contributions.

**Comment by erdi** [comment]
FWIW and a bit of context, @ArpitLT reached out about this in the past and I explained to him that it's not possible as it would require rewriting history which is not something that I'm willing to do.

**Comment by ghost** [comment]
Thanks guys. I will check with github support if something they could do to retain commits with current email for geb.
